### ISSUE TITLE
Getting info about a url consistently takes 5 seconds

### ISSUE THREAD

**Issue by kishaningithub** [pytube/pytube#208]
I am using python 3 and pytube 8.0.2
I am consistently observing a 5 second delay even in the fastest of internet connections.
Can this be improved ?
```bash
➜  pytube 'https://www.youtube.com/watch?v=ZVx_ELdttg8' --list | gnomon
   5.3422s   <Stream: itag="22" mime_type="video/mp4" res="720p" fps="30fps" vcodec="avc1.64001F" acodec="mp4a.40.2">
   0.0005s   <Stream: itag="43" mime_type="video/webm" res="360p" fps="30fps" vcodec="vp8.0" acodec="vorbis">
   ...

     Total   5.3550s
```

**Comment by kishaningithub** [comment]
Hi @nficano  I also ran the same test in a similar go lang lib.. The same fetch took only 600 milliseconds.. I am not sure what they are doing.. Do you think its because of the programming language ?

```
➜  bin ./ytdl -j 'https://www.youtube.com/watch?v=fdGGtzF2qQo' | jq '.title' | gnomon
   0.6065s   "Velaikkaran - Karuthavanlaam Galeejaam Lyric Video | Sivakarthikeyan, Nayanthara | Anirudh"
   0.0005s

     Total   0.6072s
```

**Comment by kishaningithub** [comment]
Also a parallel performance comparison with [go lang ytdl](https://github.com/rylio/ytdl)
Note that i am using python 3 and latest version of pytube (8.0.2)

![screen shot 2018-02-04 at 10 46 51 am](https://user-images.githubusercontent.com/763760/35774694-c009bb2c-099c-11e8-8c9d-fc4deb030ef5.png)

**Comment by diegocr** [comment]
This looks the same reported in #180 

Thinking further on it, i wonder if those slow downs are caused by the debugging calls all over the place:

https://github.com/nficano/pytube/blob/08752a90e3d391c042a5a783b9746e4fb5257fca/pytube/streams.py#L234-L240

:thinking:

**Comment by nficano** [comment]
it does take a little time to fetch the data from youtube and parse it, that's a limitation of the network connection

**Comment by apiraino** [comment]
I think this is solved in v9.2.3 (that implements pull request #290)

@nficano release on pypi?
